tests/Meilisearch.Tests/KeyTests.cs

```diff
@@ -0,0 +1,119 @@
+namespace Meilisearch.Tests
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+    using FluentAssertions;
+    using Xunit;
+
+    [Collection("Sequential")]
+    public class KeyTests : IAsyncLifetime
+    {
+        private IndexFixture fixture;
+        private MeilisearchClient client;
+
+        public KeyTests(IndexFixture fixture)
+        {
+            this.fixture = fixture;
+            this.client = fixture.DefaultClient;
+        }
+
+        public async Task InitializeAsync()
+        {
+            await this.fixture.DeleteAllIndexes(); // Test context cleaned for each [Fact]
+        }
+
+        public Task DisposeAsync() => Task.CompletedTask;
+
+        [Fact]
+        public async Task GetAllKeys()
+        {
+            var keyResponse = await this.client.GetKeysAsync();
+            var keys = keyResponse.Results;
+            keys.Count().Should().BeGreaterOrEqualTo(2);
+        }
+
+        [Fact]
+        public async Task GetOneKeyUsingKeyUid()
+        {
+            var keyResponse = await this.client.GetKeysAsync();
+            var keys = keyResponse.Results;
+            var firstKey = keys.First();
+
+            var fetchedKey = await this.client.GetKeyAsync(firstKey.KeyUid);
+
+            fetchedKey.KeyUid.Should().Equals(firstKey.KeyUid);
```

Comment:
Asking a dumb question, if we just compare the two instances they will not be equal right? (probably we will need to implement the `IEquatable`) either way, maybe is not the case to just check if the id is the same?

When I started to read multiple implementations of the same code I kind of see some potential for simplification, eg. In this case to ensure that the result is what we expect we just need to check if the response has the type we expect (`Key`) and if the `KeyUid` is what we expect, right? Otherwise, we are testing the implementation of the API (which is probably covered by tests also by the MeiliSearch core team).
I'll learn more about that (and get some references about it) and how I could express myself around that idea, but I just wanted to get some basic answers from you in advance :)

Reply:
Thank you for your detailed answer, I'll use that in other opportunities ;)